src/Assets.php

```diff
@@ -69,8 +69,8 @@ public static function register_assets() {
 		self::register_script( 'wc-price-filter', plugins_url( self::get_block_asset_build_path( 'price-filter' ), __DIR__ ), $block_dependencies );
 		self::register_script( 'wc-attribute-filter', plugins_url( self::get_block_asset_build_path( 'attribute-filter' ), __DIR__ ), $block_dependencies );
 		self::register_script( 'wc-active-filters', plugins_url( self::get_block_asset_build_path( 'active-filters' ), __DIR__ ), $block_dependencies );
-		self::register_script( 'wc-checkout', plugins_url( self::get_block_asset_build_path( 'checkout' ), __DIR__ ), $block_dependencies );
-		self::register_script( 'wc-cart', plugins_url( self::get_block_asset_build_path( 'cart' ), __DIR__ ), $block_dependencies );
+		self::register_script( 'wc-checkout-block', plugins_url( self::get_block_asset_build_path( 'checkout' ), __DIR__ ), $block_dependencies );
```

Comment:
This feels like a workaround – we should prefix all our script handles (and style handles), namespace-style, to avoid this kind of problem happening. Otherwise we're coupled to woocommerce core and our handles need to explicitly play nice.

E.g. our script handles would be `wc-blocks-checkout` `wc-blocks-all-products` etc. (or `wcblocks`).

Is there a way we can prefix all our handles automatically? If not, then I'd still suggest we manually do this - e.g. `wc-blocks-`.

Reply:
I was also thinking; why do we need to rename the blocks themselves and not just prefix/suffix only the handles? Handles are all that conflict here.

Reply:
> This feels like a workaround – we should prefix all our script handles (and style handles), namespace-style, to avoid this kind of problem happening. Otherwise we're coupled to woocommerce core and our handles need to explicitly play nice.
> 
> E.g. our script handles would be wc-blocks-checkout wc-blocks-all-products etc. (or wcblocks).

I think this is a valid point, however in some ways blocks (along with wc-admin) once merged to core _are_  (for all intents and purposes) _part_ of core.  The existing handles have been in place for a while without issue so I think it'd be unwise to unilaterally change them.  I do think we need to be more aware of where handles might conflict with core though for new ones we add (and adjust handle names accordingly)

> Is there a way we can prefix all our handles automatically? If not, then I'd still suggest we manually do this - e.g. wc-blocks-.

Yea I think we should, I'd like to do that as a part of work scoped for #973 where more than just handle naming is automated.  We have to be careful about renaming too much here because there may be reliance on the existing handles we don't know about.

> why do we need to rename the blocks themselves and not just prefix/suffix only the handles?

Good question.  I'm thinking ahead to when we might automate this and the handle would be defined as a part of the block config (which is currently _somewhat_ what happens in the php block classes right now).  If you think `block` is redundant for usage in the block name, I'm open to suggestions on how we could improve that?

### Next steps

We have a larger issue in my mind (that the two of you helped reinforce by your feedback) in that our block registration php side has a _lot_ of moving parts and places that need manually updated to ensure blocks are correctly registered, their assets are correctly loaded (only when needed), and the necessary data a block needs is pre-loaded.  On top of that we have to be aware of potential conflicts with WooCommerce core (and/or extensions as well!).  To a large degree, this is why I created #973 because I think the loading and initialization process could use some improving to reduce the chance for human error and automate as much as we can.

For the purpose of this pull, I'd like to avoid spending too much time implementing that automation that might need redoing when we get to #973 (which I'm thinking of tackling in the next cooldown). So would it be okay with you guys to merge this as is and leave the other things for 973?

Reply:
Works for me – sounds like you have a good handle on the considerations here.

Reply:
> If you think block is redundant for usage in the block name, I'm open to suggestions on how we could improve that?

Suggestion - don't use `block` for block slug. If there's a prefix for the plugin / code base the blocks come from – i.e. `woocommerce-blocks`, this repo, this plugin – then that could have block in it. Examples: `woocommerce-blocks/all-products`, `woocommerce-blocks/cart`, etc :)

Reply:
> this repo, this plugin – then that could have block in it. Examples: woocommerce-blocks/all-products, woocommerce-blocks/cart, etc :)

Again though, this plugin is included bundled in core so I think we should treat prefixes/suffixes a bit differently for things like script handles.   Let's continue the conversation in this pull and we can create other pulls/issues to address if necessary (or add to #973) but for now I think I'll pull the trigger on merging this.